content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,72 +1,48 @@
-
 This is an article about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
-
 ### Introduction
 Socket programming is a way of connecting two computers over a network to allow the exchange of information.The connection can be through **connection-oriented protocol** or **connectionless**. In our case, we will be using  TCP/IP a connection-oriented protocol.
-
 In connection-oriented protocols, two computers must establish a connection before transferring data. Whereas in connectionless, you do not need every packet to arrive at its desired destination. This is made possible by UDP(User Datagram Protocol)
-
-We will be using the Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from socket connection.To establish this connection,the machines must have each other's network information.
-
+We will be using the Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from the socket connection. To establish this connection, the machines must have each other's network information.
 Table of contents:
 - [Definition-of-a-Socket](#definition-of-a-socket)
```

Comment:
> - \[Definition-of-a-Socket\]\(#definition-of-a-socket)
> - \[Server-Side-Programming\ ](#server-side-programming)
> - \[Client-Side-Programming\](#client-side-programming)
> - \[How-to-Run-the-Programs-on-different-applications\](#how-to-run-the-programs-on-diffrent-applications)
> - \[Conclusion\](#conclusion)

You do not need to put *-*'s between the words in the square brackets (*[* and *]*).  So change it to:

> - \[Definition of a Socket\](#definition-of-a-socket)
> - \[Server Side Programming\](#server-side-programming)
> - \[Client Side Programming\](#client-side-programming)
> - \[How to Run the Programs on different applications\](#how-to-run-the-programs-on-diffrent-applications)
> - \[Conclusion\](#conclusion)